pkg/internal/affinityassistant/affinityassistant_types_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package affinityassistant
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	cfgtesting "github.com/tektoncd/pipeline/pkg/apis/config/testing"
+	"github.com/tektoncd/pipeline/test/diff"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+)
+
+func Test_GetAffinityAssistantBehavior(t *testing.T) {
+	tcs := []struct {
+		name      string
+		configMap map[string]string
+		expect    AffinityAssitantBehavior
+	}{{
+		name: "affinity-assistant-enabled",
```

Comment:
In the latest combination chart, this combination is invalid and we have validated this in the feature_flags.go, so we won't encounter such scenario at this point

Reply:
I think it could make sense for `GetAffinityAssistantBehavior` to return an error for invalid combinations. That way if a new value is added for the feature flag and we accidentally forget to update that function, we will fail fast rather than returning just an empty string. It would also help catch issues like accidentally removing validation.

Reply:
Make sense. I added an error as a return type in the function. However, it seems I cannot write a UT to cover the invalid scenario as I cannot pass in a ConfigMap in the test like:
```
configMap: map[string]string{
			"disable-affinity-assistant": "true",
			"coscheduling":               "unknown value",
		}
```

since parsing this configMap to `feature flags` will during the validation.

Reply:
Hmm, I tried to add invalid combinations to the test case, but it looks like I cannot validate the error from `SetFeatureFlags` as `s.OnConfigChanged` does not return an error. It directly fails when passing in invalid values

Reply:
ah ok, sg